src/core/datatypes/phylogenetics/substitution/MixtureModel.h

```diff
@@ -0,0 +1,74 @@
+#ifndef MIXTURE_MODEL_H
+#define MIXTURE_MODEL_H
+
+#include <stddef.h>
+#include <vector>
+#include <tuple>
+#include <optional>
+
+//#include "MatrixReal.h"
+#include "MemberObject.h"  // For member functions.
+#include "Cloneable.h"
+#include "TransitionProbabilityMatrix.h"
+
+namespace RevBayesCore {
+
+    class Tree;
+
+    /**
+     * @brief Abstract mixture model class.
+     *
+     * Derived models that depend on the branch lengths will additionally need to
+     * incorporate a reference to the tree.
+     *
+     * @copyright Copyright 2022-
+     * @author Benjamin Redelings
+     * @since 2022-09-22, version 1.0
+     */
+
+    class MixtureModel: public Cloneable, public MemberObject< RbVector<RbVector<RbVector<double>>> >
```

Comment:
Just my personal view:
Could you rename these into SubstitutionMixtureModel? I think that is more appropriate. Note that we have other types of mixture models too, e.g., https://github.com/revbayes/revbayes/blob/7b872a7095c5342297ad4a1f254b29d81cc68040/src/core/distributions/AnalyticalMixtureDistribution.h#L26

Reply:
Yeah, good point.